File: 3069-distribute-elements-into-two-arrays-i/3069-distribute-elements-into-two-arrays-i.cpp
class Solution {
public:
    bool check (vector<int>& arr1,vector<int>& arr2) {
        if (arr1.back() >= arr2.back()) return true;

        return false;
    }

    vector<int> resultArray(vector<int>& nums) {
        vector<int> arr1;    
        vector<int> arr2;

        arr1.push_back(nums[0]);    
        arr2.push_back(nums[1]);

        bool flag = check(arr1, arr2);

        for (int i=2; i<nums.size(); i++) {
            if (flag) {
                arr1.push_back(nums[i]);
            } else {
                arr2.push_back(nums[i]);
            }

            flag = check(arr1, arr2);
        }

        int a=0; 

        for (int i=0; i<arr1.size(); i++) {
            nums[a++] = arr1[i];
        }
        for (int i=0; i<arr2.size(); i++) {
            nums[a++] = arr2[i];
        }

        return nums;
    }
};

File: 3069-distribute-elements-into-two-arrays-i/3069-distribute-elements-into-two-arrays-i.java
class Solution {
    boolean check (ArrayList<Integer> arr1, ArrayList<Integer> arr2) {
        if (arr1.get(arr1.size() - 1) >= arr2.get(arr2.size() - 1)) return true;

        return false;
    }

    public int[] resultArray(int[] nums) {
        ArrayList<Integer> arr1 = new ArrayList<>();
        ArrayList<Integer> arr2 = new ArrayList<>();
    
        arr1.add(nums[0]);
        arr2.add(nums[1]);

        boolean flag = check(arr1, arr2);

        for (int i=2; i<nums.length; i++) {
            if (flag) {
                arr1.add(nums[i]);
            } else {
                arr2.add(nums[i]);
            }

            flag = check(arr1, arr2);
        }

        int a=0; 

        for (int i=0; i<arr1.size(); i++) {
            nums[a++] = arr1.get(i);
        }
        for (int i=0; i<arr2.size(); i++) {
            nums[a++] = arr2.get(i);
        }

       return nums;
    }
}